db_sql.go

```diff
@@ -174,20 +174,22 @@ func (x *sqlUserStoreDB) CreateIdentity(email, username, firstname, lastname, mo
 
 		// Get user id
 		var userId int64
-		row = tx.QueryRow(`SELECT userid FROM authuserstore WHERE email = $1`, email)
+		row := tx.QueryRow(`SELECT userid FROM authuserstore WHERE email = $1`, email)
 		if scanErr := row.Scan(&userId); scanErr != nil {
 			tx.Rollback()
 			return NullUserId, scanErr
 		}
 
-		// Insert into auth user
-		if _, eCreateAuthUser := tx.Exec(`INSERT INTO authuserpwd (userid, password) VALUES ($1, $2)`, userId, hash); eCreateAuthUser != nil {
-			//fmt.Printf("Insert into authuserpwd failed because: %v\n", eCreateAuthUser)
-			if strings.Index(eCreateAuthUser.Error(), "duplicate key") != -1 {
-				eCreateAuthUser = ErrIdentityExists
+		if !ldapUsed {
```

Comment:
Instead of referring to a global flag here, sqlUserStoreDB could use a member variable called 'enableAuthenticator'. Either way - no globals like that.